libtiledbsoma/src/soma/soma_array.cc

```diff
@@ -1422,11 +1422,6 @@ void SOMAArray::_set_current_domain_from_shape(
     // Variant-indexed dataframes must use a separate path
     _check_dims_are_int64();
 
-    if (_get_current_domain().is_empty()) {
```

Comment:
This was wrong, and not caught sooner b/c I didn't loop over both false & true in the unit-test code.

The is-empty check is done in both callers of this method.